src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,44 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Pair<K,V>[] data;
+    private int size;
+    
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        data = new Pair[INITIAL_CAPACITY];
+        size = 0;
+    }
+    
+    private boolean keysEqual(K a, K b) {
+        return a == b || (a != null && a.equals(b));
```

Comment:
```suggestion
        return a == b || a != null && a.equals(b);
```
redundant brackets, see logical operator priority